internal/dbadapter/db_adapter.go

```diff
@@ -377,66 +352,53 @@ func (a *StandardAdapter) JobGetAvailableTx(ctx context.Context, tx pgx.Tx, queu
 	return jobs, nil
 }
 
-func (a *StandardAdapter) JobSetCancelledIfRunning(ctx context.Context, id int64, finalizedAt time.Time, err []byte) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
-
-	return a.queries.JobSetCancelledIfRunning(ctx, a.executor, dbsqlc.JobSetCancelledIfRunningParams{
-		ID:          id,
-		Error:       err,
-		FinalizedAt: finalizedAt,
-	})
+// JobSetStateParams are parameters to update the state of a currently running
+// job. Use one of the constructors below to ensure a correct combination of
+// parameters.
+type JobSetStateParams struct {
+	ID          int64
+	errData     []byte
+	finalizedAt *time.Time
+	maxAttempts *int
+	scheduledAt *time.Time
+	state       dbsqlc.JobState
 }
 
-func (a *StandardAdapter) JobSetCompletedIfRunning(ctx context.Context, job JobToComplete) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
-
-	return a.queries.JobSetCompletedIfRunning(ctx, a.executor, dbsqlc.JobSetCompletedIfRunningParams{
-		ID:          job.ID,
-		FinalizedAt: job.FinalizedAt.UTC(),
-	})
+func JobSetStateCancelled(id int64, finalizedAt time.Time, errData []byte) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, errData: errData, finalizedAt: &finalizedAt, state: dbsqlc.JobStateCancelled}
```

Comment:
See these new params constructors, but the general idea is that they help ensure that only valid parameter combinations are generated for the update query.